src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -550,15 +556,18 @@ public NativeObjectWrapper(IntPtr externalComObject, IntPtr inner, ComWrappers c
 
             public virtual void Release()
             {
-                if (_comWrappers != null)
+                if (_maybeCached)
                 {
-                    _comWrappers.RemoveRCWFromCache(_externalComObject, _proxyHandle);
-                    _comWrappers = null;
+                    if (ProxyHandle.IsAllocated)
+                    {
+                        _comWrappers.RemoveRCWFromCache(_externalComObject, ProxyHandle);
+                    }
+                    _maybeCached = false;
                 }
 
-                if (_proxyHandle.IsAllocated)
+                if (ProxyHandle.IsAllocated)
                 {
-                    _proxyHandle.Free();
+                    ProxyHandle.Free();
```

Comment:
This will operate on clone of GCHandle (even before this change). The actual field is going to be a dangling pointer after the Free. Is it a problem?